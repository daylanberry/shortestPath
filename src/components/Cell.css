.image {
  height: 100px;
  width: 150px;
  background-size: contain;
  background-repeat:no-repeat;
  border-radius: 50px;
  position: absolute;
  margin-right: 10px;
}

.meeple {
  height: 50px;
  margin-left: 50px;
  position: absolute;
}
.con {
  height: 200px;
}


.blank {
  height: 150px;
}

.meep {
  height: 20px;
}