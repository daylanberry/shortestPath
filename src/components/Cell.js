import React from 'react'
import downTown from '../assets/downtown-Reno.jpg'
import lakeTahoe from '../assets/lake-tahoe.jpg'
import legend from '../assets/legend.jpg'
import house from '../assets/house.jpeg'
import { ReactComponent as Meeple } from '../assets/meeple.svg'

import './Cell.css'

const Cell = ({ x, y, dt, legends, truckee, unr, home }) => {


  if (dt[0] === x && dt[1] === y) {
    return <div> <img className='image' src={downTown}/>C</div>

  } else if (x === 4 && y === 4) {
    return <div> <img className='image' src={lakeTahoe}/>F</div>

  } else if (unr[0] === x && unr[1] === y) {
    return <div > < img className = 'image' src={
      'https://upload.wikimedia.org/wikipedia/en/5/55/UNR_Campus_North120520.jpg'
    }/>B</div >

  } else if (legends[0] === x && legends[1] === y) {
    return <div> <img className='image' src={legend}/>E</div>

  } else if (truckee[0] === x && truckee[1] === y) {
    return <div> <img className='image' src={'https://i.pinimg.com/originals/93/c7/f7/93c7f77f60f2b27e3a32b179731b34a3.jpg'}/>D</div>

  } else if (home[0] === x && home[1] === y) {
    return <div className='image'>
        <img className='image' src={'https://media.graytvinc.com/images/690*407/paddock+house+reno1.jpg'}/>
        A
      </div>
  }

  else {
    return <div className='blank'></div>
  }
}

export default Cell