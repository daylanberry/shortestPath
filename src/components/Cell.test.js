import React from 'react';
import Cell from './Cell';
import renderer from 'react-test-renderer'



it('should return desired cell for house position', () => {
  const properties = {
    x: 0,
    y: 0,
    home: [0, 0],
    dt: [1, 2],
    legends: [1, 4],
    truckee: [4, 2],
    unr: [2, 0]
  }

  const cellComponent = renderer.create(<Cell {...properties}/>).toJSON()
  expect(cellComponent).toMatchSnapshot()

  expect(1 + 1).toEqual(2)
})

it('should return desired cell for tahoe position', () => {
  const properties = {
    x: 4,
    y: 4,
    home: [0, 0],
    dt: [1, 2],
    legends: [1, 4],
    truckee: [4, 2],
    unr: [2, 0]
  }

  const cellComponent = renderer.create(<Cell {...properties}/>).toJSON()
  expect(cellComponent).toMatchSnapshot()

  expect(1 + 1).toEqual(2)
})