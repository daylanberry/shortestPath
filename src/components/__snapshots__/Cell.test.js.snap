// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return desired cell for house position 1`] = `
<div
  className="image"
>
  <img
    className="image"
    src="https://media.graytvinc.com/images/690*407/paddock+house+reno1.jpg"
  />
  A
</div>
`;

exports[`should return desired cell for tahoe position 1`] = `
<div>
   
  <img
    className="image"
    src="lake-tahoe.jpg"
  />
  F
</div>
`;
